scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_view.py

```diff
@@ -8,34 +8,63 @@
 from os import path
 
 from mantidqt.utils.qt import load_ui
+from Engineering.gui.engineering_diffraction.tabs.common import path_handling
 
 Ui_data, _ = load_ui(__file__, "data_widget.ui")
 
 
+class FileFilterProxyModel (QtCore.QSortFilterProxyModel):
+    text_filter = None
+
+    def __init__(self, parent=None):
+        super(FileFilterProxyModel, self).__init__(parent)
+
+    def filterAcceptsRow(self, source_row, source_parent):
+        model = self.sourceModel()
+        index0 = model.index(source_row, 0, source_parent)
+        fname = model.fileName(index0)
+        if self.text_filter is None:
+            return True
+        else:
+            return model.isDir(index0) or (fname.find(self.text_filter) >= 0)
+
+
 class FittingDataView(QtWidgets.QWidget, Ui_data):
     sig_enable_load_button = QtCore.Signal(bool)
     sig_enable_inspect_bg_button = QtCore.Signal(bool)
+    proxy_model = None
 
     def __init__(self, parent=None):
         super(FittingDataView, self).__init__(parent)
         self.setupUi(self)
         # file finder
+        self.finder_data.readSettings(path_handling.INTERFACES_SETTINGS_GROUP + '/' + path_handling.ENGINEERING_PREFIX)
```

Comment:
This change just makes the browse dialog on the Fitting tab remember the folder that the user last selected a file from. So it will only return to the focussed files folder if you've previously selected a file from there
Note that if a path has been automatically filled in by the user having run Focus just before then this takes precedence over the path read from the settings and it will also gets saved as the last folder when you shut down the UI.